proxy/src/bind.rs

```diff
@@ -170,7 +196,14 @@ where
             self.executor.clone(),
         );
 
-        let proxy = self.sensors.http(self.req_ids.clone(), client, &client_ctx);
+        let sensors = self.sensors.http(
+            self.req_ids.clone(),
+            client,
+            &client_ctx
+        );
+
+        // Rewrite the HTTP/1 URI, if necessary.
+        let proxy = ReconstructUri::new(sensors);
```

Comment:
When is it necessary to rewrite requests?

I don't think we should be mutating requests at all. If we're doing so, let's be very clear about explaining why and when we're doing so.

So, at the least, please add some clarity in this comment.

Modulo that can someone give me the summary of why request rewrites are necessary?

Reply:
Request rewriting was not added in this PR (I believe it was added in https://github.com/runconduit/conduit/commit/54aef56e25f132b8e8e7d6ab2eb356ec4d7be890#diff-9b5ffd7799ff3ebde74fba6d9c3a35dc). All this PR does is move _where_ the call to `reconstruct_uri` takes place.

I'm not aware of all the rationale behind why we do this, although I know that removing it currently breaks some of the telemetry unit tests. I think @seanmonstar has more insight into why the request rewriting code was added.

Reply:
ok, we can revisit that in another issue.  It would be helpful to add something here getting clearer on "if necessary", but not a blocker.

Reply:
At the very least, I can definitely change that comment to give a better description of _what_ rewriting takes place, if not why.